local.go

```diff
@@ -83,40 +82,42 @@ func Local(f Function, initX []float64, settings *Settings, method Method) (*Res
 	}
 
 	// actually perform the minimization
-	status, err := minimize(settings, location, method, funcInfo, stats, funcs, optLoc, startTime)
+	status, err := minimize(settings, &loc, method, funcInfo, stats, funcs, startTime)
+
+	var optLoc Location
+	copyLocation(&optLoc, loc)
 
 	// cleanup at exit
 	if settings.Recorder != nil && err == nil {
-		err = settings.Recorder.Record(*optLoc, NoEvaluation, PostIteration, stats)
+		err = settings.Recorder.Record(optLoc, NoEvaluation, PostIteration, stats)
 	}
 	stats.Runtime = time.Since(startTime)
 	return &Result{
 		Stats:    *stats,
-		Location: *optLoc,
+		Location: optLoc,
```

Comment:
I agree with changing all Location to *Location. It is a big change, though, so I propose to split this PR into several smaller ones.